src/modules/endpoint/sagas/loadCityContent.js

```diff
@@ -2,26 +2,25 @@
 
 import type { Saga } from 'redux-saga'
 import { all, call } from 'redux-saga/effects'
-import MemoryDatabase from '../MemoryDatabase'
-import MemoryDatabaseContext from '../MemoryDatabaseContext'
+import DataContainer from '../DataContainer'
 import loadLanguages from './loadLanguages'
 import loadCategories from './loadCategories'
 import loadEvents from './loadEvents'
 import fetchResourceCache from './fetchResourceCache'
 
-export default function * loadCityContent (database: MemoryDatabase, newCity: string, newLanguage: string): Saga<void> {
-  if (database.hasContext(new MemoryDatabaseContext(newCity, newLanguage))) {
-    return // All data is already in the database
+export default function * loadCityContent (dataContainer: DataContainer, newCity: string, newLanguage: string): Saga<void> {
+  if (dataContainer.hasContext(newCity, newLanguage)) {
+    return
   }
```

Comment:
I think we should remove this check. It shouldn't harm if we just set the context on the datacontainer even though it's already set.
(Hence we should also remove the function hasContext from dataContainer)